File: meditime_frontend/lib/models/doctor_model.dart
import 'user_model.dart';

class Doctor {
  final int id;
  final int idUser;
  final String specialite;
  final String diplomes;
  final String numeroInscription;
  final String hopital;
  final String adresseConsultation;
  final double note; // jamais null
  final DateTime createdAt;
  final User? user;

  // Nouveaux champs
  final int? patientsExamined;
  final int? experienceYears;
  final int? pricePerHour;
  final String? description;

  Doctor({
    required this.id,
    required this.idUser,
    required this.specialite,
    required this.diplomes,
    required this.numeroInscription,
    required this.hopital,
    required this.adresseConsultation,
    required this.note,
    required this.createdAt,
    this.user,
    this.patientsExamined,
    this.experienceYears,
    this.pricePerHour,
    this.description,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['id'],
      idUser: json['idUser'],
      specialite: json['specialite'],
      diplomes: json['diplomes'],
      numeroInscription: json['numero_inscription'],
      hopital: json['hopital'],
      adresseConsultation: json['adresse_consultation'],
      note: (json['note'] as num?)?.toDouble() ?? 0,
      createdAt: DateTime.parse(json['created_at']),
      user: json['user'] != null ? User.fromMap(json['user']) : null,
      patientsExamined: json['patientsExamined'],
      experienceYears: json['experienceYears'],
      pricePerHour: json['pricePerHour'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'idUser': idUser,
      'specialite': specialite,
      'diplomes': diplomes,
      'numero_inscription': numeroInscription,
      'hopital': hopital,
      'adresse_consultation': adresseConsultation,
      'note': note,
      'created_at': createdAt.toIso8601String(),
      'patientsExamined': patientsExamined,
      'experienceYears': experienceYears,
      'pricePerHour': pricePerHour,
      'description': description,
      'user': user?.toMap(),
    };
  }
}

File: meditime-backend/src/models/doctor_model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');
const User = require('./user_model');

const Doctor = sequelize.define('Doctor', {
  id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
  idUser: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: { model: 'users', key: 'idUser' }
  },
  specialite: { type: DataTypes.STRING(100), allowNull: false },
  diplomes: { type: DataTypes.TEXT, allowNull: false },
  numero_inscription: { type: DataTypes.STRING(100), allowNull: false },
  hopital: { type: DataTypes.STRING(150), allowNull: false },
  adresse_consultation: { type: DataTypes.STRING(255), allowNull: false },
  note: { type: DataTypes.FLOAT, allowNull: false, defaultValue: 0 },
  created_at: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
  patientsExamined: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
  experienceYears: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
  pricePerHour: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
  description: { type: DataTypes.TEXT, allowNull: true }
}, {
  tableName: 'doctor',
  timestamps: false
});

Doctor.belongsTo(User, { foreignKey: 'idUser', as: 'user' });

module.exports = Doctor;